### DISCUSSION TITLE
GPX export

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#89] [Q&A] [answered]
When exporting a route the checkboxes for waypoints and route are greyed out and can't be selected. Why?

**Comment by WolfgangZ42** [comment]
I guess you have selected the navigation option:
<img width="200" src="https://user-images.githubusercontent.com/121806238/226268376-97d5e4b1-726c-4eff-b750-f6545f25ba0d.png"/>  

If exporting a "normal" gpx file it looks like this:
<img width="200" src="https://user-images.githubusercontent.com/121806238/226268471-6ade361e-0a61-4f71-ad40-bebbddd2291c.png"/>

**Comment by DolomitesTom** [comment]
yes, but I don't understand. What is a "normal" gpx file ? I only have the option GPX or Kurviger. In K2 I can choose GPX offline and can select any of the checkboxes, nothing is deactivated

**Reply by 0709wiwiwi** [reply]
Export .gpx 
**SET** navigation.
The gpx track export is enriched. (Similar to .kurviger file)
"Contains turns&via&shaping. Allows so a 100% faithfull (re)import. 
The **gpx navigation track** allows immediate navigation after import.

**UNSET** navigation. 
You are able to set export.
- gpx (trk) track (naked)  (what you call a normal track)
- gpx (rte) route
- gpx (wpt)

The Kurviger app and other recognise the **gpx navigation track** as a standard simple track.
The extra (enriched info) only is recognised by Cruiser and by Locus. (Compatible)

To study what that **gpx navigation track** contains as an extra ? 
Open by 'freeware" gpx editor (windows) and see the trackpoint list.

https://youtu.be/PplQPxxNmHE

**Comment by DolomitesTom** [comment]
sorry, I still don't understand. How can I active the checkboxes for waypoints and route in Cruiser gpx export. In K2 I could choose wheather I want to export a track or a route with waypoints. Can Cruiser offer the same? When I import the Cruiser gpx in the BMW Connector App in order to check traffic on the route, all my waypoints are lost, because it is a track without waypoints.

**Reply by WolfgangZ42** [reply]
> How can I active the checkboxes for waypoints and route in Cruiser gpx export.

just uncheck "Navigation" and you can check "Route" and "Waypoints"

**Reply by DolomitesTom** [reply]
ok now I understood, thanks 🙂

**Comment by devemux86** [comment, marked as answer]
GPX navigation track is an **offline** GPX format that stores everything: route path, navigation instructions, etc.
It is used by BRouter, Locus, [RouteYou](https://www.routeyou.com/), ...

Kurviger is a similar format I wrote for K1 / K2 so can load **offline** the routes.
Then we offered it in the app and website export / import dialogs.

---

Cruiser can export and import **offline** GPX navigation tracks.

GPX navigation track can also store all the Kurviger routing parameters,
so you can convert from one format to another while keeping all options.

_And everything is done **offline**._

**Reply by DolomitesTom** [reply]
Tank you
